use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/service/TimeTableResource.java

```diff
@@ -0,0 +1,165 @@
+
+    private ConcurrentMap<String, Job> jobs = new ConcurrentHashMap<>();
+
+    // Workaround to make Quarkus CDI happy. Do not use.
+    public TimeTableResource() {
+        this.solverManager = null;
+    }
+
+    @Inject
+    public TimeTableResource(SolverManager<TimeTable, String> solverManager) {
+        this.solverManager = solverManager;
+    }
+
+    @Operation(summary = "List the job IDs of all submitted timetables.")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "List of all job IDs.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(type = SchemaType.ARRAY, implementation = String.class))) })
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<String> list() {
+        return jobs.keySet().stream().toList();
+    }
+
+    @Operation(summary = "Submit a timetable to start solving as soon as CPU resources are available.")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200",
+                    description = "The job ID. Use that ID to get the solution with the other methods.",
+                    content = @Content(mediaType = MediaType.TEXT_PLAIN, schema = @Schema(implementation = String.class))) })
+    @POST
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces(MediaType.TEXT_PLAIN)
+    public String solve(TimeTable problem) {
+        String jobId = UUID.randomUUID().toString();
+        jobs.put(jobId, Job.newTimeTable(problem));
+        solverManager.solveAndListen(jobId,
+                jobId_ -> jobs.get(jobId).timeTable,
+                solution -> jobs.put(jobId, Job.newTimeTable(solution)),
+                (jobId_, exception) -> jobs.put(jobId, Job.error(jobId_, exception)));
+        return jobId;
+    }
+
+    @Operation(
+            summary = "Get the solution and score for a given job ID. This is the best solution so far, as it might still be running or not even started.")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "The best solution of the timetable so far.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = TimeTable.class))),
+            @APIResponse(responseCode = "404", description = "No timetable found.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = ErrorInfo.class))),
+            @APIResponse(responseCode = "500", description = "Exception during solving a timetable.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = ErrorInfo.class)))
+    })
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{jobId}")
+    public Response getTimeTable(
```

Comment:
The method could return the `TimeTable` directly, but error handling would require `ExceptionMappers` to return the proper code and message to the client.
I picked this approach for the simplicity, which is good for quickstarts).